src/main/java/com/configcat/FormattableLogMessageWithKeySet.java

```diff
@@ -0,0 +1,27 @@
+package com.configcat;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+class FormattableLogMessageWithKeySet extends FormattableLogMessage {
+
+    private String cachedMessage;
+    private final String message;
+    private final Object[] args;
```

Comment:
Since these fields are already stored by the superclass, storing them here too is redundant, thus, wastes 16 bytes of memory per instance on an x64 platform.

Let's remove these fields from here and just mark the ones in the superclass as protected.

```suggestion
```